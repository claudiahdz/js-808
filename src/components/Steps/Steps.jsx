import React from 'react'
import PropTypes from 'prop-types'

const Steps = ({ status, currentStep }) => {
  let active
  let steps = []

  Array.from(Array(16)).map((elem, i) => {
    active = (currentStep === i) && (status === 'play')
    return steps.push(
      <span key={i} className={active ? 'active' : ''}}>
        {i+1}
      </span>
    )
  })

  return (
    <div className="steps">
      {steps}
    </div>
  )
}

Steps.propTypes = {
  status: PropTypes.string.isRequired,
  currentStep: PropTypes.number.isRequired
}

export default Steps
