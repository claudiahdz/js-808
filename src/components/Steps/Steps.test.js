import React from 'react'
import { shallow } from 'enzyme'
import Steps from './Steps'

describe('Steps', () => {
  const defaultProps = {
    status: 'stop',
    currentStep: 0
  }

  it('renders the component correctly', () => {
    const steps = shallow(<Steps {...defaultProps} />)
    expect(steps).toMatchSnapshot()
  })
})

