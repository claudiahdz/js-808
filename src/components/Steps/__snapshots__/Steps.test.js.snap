// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps renders the component correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="steps"
>
    <span
        className="false"
    >
        1
    </span>
    <span
        className="false"
    >
        2
    </span>
    <span
        className="false"
    >
        3
    </span>
    <span
        className="false"
    >
        4
    </span>
    <span
        className="false"
    >
        5
    </span>
    <span
        className="false"
    >
        6
    </span>
    <span
        className="false"
    >
        7
    </span>
    <span
        className="false"
    >
        8
    </span>
    <span
        className="false"
    >
        9
    </span>
    <span
        className="false"
    >
        10
    </span>
    <span
        className="false"
    >
        11
    </span>
    <span
        className="false"
    >
        12
    </span>
    <span
        className="false"
    >
        13
    </span>
    <span
        className="false"
    >
        14
    </span>
    <span
        className="false"
    >
        15
    </span>
    <span
        className="false"
    >
        16
    </span>
</div>,
  "nodes": Array [
    <div
      className="steps"
>
      <span
            className="false"
      >
            1
      </span>
      <span
            className="false"
      >
            2
      </span>
      <span
            className="false"
      >
            3
      </span>
      <span
            className="false"
      >
            4
      </span>
      <span
            className="false"
      >
            5
      </span>
      <span
            className="false"
      >
            6
      </span>
      <span
            className="false"
      >
            7
      </span>
      <span
            className="false"
      >
            8
      </span>
      <span
            className="false"
      >
            9
      </span>
      <span
            className="false"
      >
            10
      </span>
      <span
            className="false"
      >
            11
      </span>
      <span
            className="false"
      >
            12
      </span>
      <span
            className="false"
      >
            13
      </span>
      <span
            className="false"
      >
            14
      </span>
      <span
            className="false"
      >
            15
      </span>
      <span
            className="false"
      >
            16
      </span>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Steps
        currentStep={0}
        status="stop"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "currentStep": 0,
          "status": "stop",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="steps"
>
          <span
                    className="false"
          >
                    1
          </span>
          <span
                    className="false"
          >
                    2
          </span>
          <span
                    className="false"
          >
                    3
          </span>
          <span
                    className="false"
          >
                    4
          </span>
          <span
                    className="false"
          >
                    5
          </span>
          <span
                    className="false"
          >
                    6
          </span>
          <span
                    className="false"
          >
                    7
          </span>
          <span
                    className="false"
          >
                    8
          </span>
          <span
                    className="false"
          >
                    9
          </span>
          <span
                    className="false"
          >
                    10
          </span>
          <span
                    className="false"
          >
                    11
          </span>
          <span
                    className="false"
          >
                    12
          </span>
          <span
                    className="false"
          >
                    13
          </span>
          <span
                    className="false"
          >
                    14
          </span>
          <span
                    className="false"
          >
                    15
          </span>
          <span
                    className="false"
          >
                    16
          </span>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="steps"
>
          <span
                    className="false"
          >
                    1
          </span>
          <span
                    className="false"
          >
                    2
          </span>
          <span
                    className="false"
          >
                    3
          </span>
          <span
                    className="false"
          >
                    4
          </span>
          <span
                    className="false"
          >
                    5
          </span>
          <span
                    className="false"
          >
                    6
          </span>
          <span
                    className="false"
          >
                    7
          </span>
          <span
                    className="false"
          >
                    8
          </span>
          <span
                    className="false"
          >
                    9
          </span>
          <span
                    className="false"
          >
                    10
          </span>
          <span
                    className="false"
          >
                    11
          </span>
          <span
                    className="false"
          >
                    12
          </span>
          <span
                    className="false"
          >
                    13
          </span>
          <span
                    className="false"
          >
                    14
          </span>
          <span
                    className="false"
          >
                    15
          </span>
          <span
                    className="false"
          >
                    16
          </span>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Steps
    currentStep={0}
    status="stop"
/>,
}
`;
